include/RAJA/policy/cuda/MemUtils_CUDA.hpp

```diff
@@ -291,6 +294,184 @@ cudaDeviceProp& device_prop()
   return prop;
 }
 
+
+struct CudaFixedMaxBlocksData
+{
+  int multiProcessorCount;
+  int maxThreadsPerMultiProcessor;
+};
+
+RAJA_INLINE
+size_t cuda_max_blocks(size_t block_size)
+{
+  static CudaFixedMaxBlocksData data = {-1, -1};
```

Comment:
would it be better to use a constexpr variable or macro constant, such as RAJA_SIZE_UNDEFINED (for the macro case, similarly w/o the RAJA prefix for the variable), instead of sprinkling '-1' in the code?

Reply:
Probably, this is only used locally to indicate that we don't yet have data.

Reply:
It's your call. I think it makes the code more easy to understand.

Reply:
Added an issue #1500

Reply:
Works for me.

Reply:
I went ahead and named the -1s and fixed the race conditions.